src/pks/flow/richards_pk.cc

```diff
@@ -418,13 +420,28 @@ void Richards::SetupPhysicalEvaluators_(const Teuchos::Ptr<State>& S) {
     auto& wrm_plist = S->GetEvaluatorList(sat_key_);
     wrm_plist.setParameters(plist_->sublist("water retention evaluator"));
     wrm_plist.set("field evaluator type", "WRM");
+
+    
+    if (!S->HasFieldEvaluator(suction_key_)) {
+      Teuchos::ParameterList wrm_plist(plist_->sublist("water retention evaluator"));
+      wrm_plist.setName(suction_key_);
+      wrm_plist.set("field evaluator type", "WRM suction head");
+      S->FEList().set(suction_key_, wrm_plist);
+
+      //-- suction
+      S->RequireField(suction_key_)->SetMesh(mesh_)->SetGhosted()
+        ->AddComponent("cell", AmanziMesh::CELL, 1);
+      S->RequireFieldEvaluator(suction_key_);
+    }
```

Comment:
Wait wait, I don't like this.  What is suction, and why is every Richards PK going to have to add a new vector and new evaluator for it?  Nowhere else in richards_* is this key used, so I don't understand what you are doing with it.  If you need it for another PK, please add it there, not in Richards PK.